src/main/java/frc/robot/subsystems/ArmSubsystem.java

```diff
@@ -0,0 +1,160 @@
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import frc.lib.lib2706.SubsystemChecker;
+import frc.lib.lib2706.SubsystemChecker.SubsystemType;
+import frc.robot.Config;
+import frc.lib.lib2706.ErrorCheck;
+import frc.lib.lib2706.ProfiledPIDFFController;
+
+public class ArmSubsystem extends SubsystemBase{
+     private static ArmSubsystem instance = null; //static object that contains all movement controls
+
+     private static final MotorType motorType = MotorType.kBrushless; //defines brushless motortype
+     public final CANSparkMax m_arm; //bottom SparkMax motor controller
+     
+     //network table entry
+     private final String m_tuningTable = "Arm/ArmTuning";
+     private final String m_dataTable = "Arm/ArmData";
+  
+     //network table entries 
+     private DoubleEntry m_armPSubs;
+     private DoubleEntry m_armISubs;
+     private DoubleEntry m_armDSubs;
+     private DoubleEntry m_armIzSubs;
+     private DoubleEntry m_armFFSubs;
+     private DoublePublisher m_armSetpointPub;   
+     private DoublePublisher m_armVelPub;
+     private DoubleEntry m_armMomentToVoltage;
+     private DoublePublisher m_armFFTestingVolts;
+     private DoubleEntry m_armOffset;
+     private DoublePublisher m_targetAngle;
+     private DoublePublisher m_armPosPub;
+
+    //for  arm ff
+    private DoubleSubscriber momentToVoltageConversion;
+    private double m_VoltageConversion;
+
+    //spark absolute encoder
+    SparkAbsoluteEncoder m_absEncoder;  
+    //embedded relative encoder
+    //private RelativeEncoder m_Encoder;
+    public SparkPIDController m_pidControllerArm;  
+
+    ProfiledPIDFFController m_profiledFFController = new ProfiledPIDFFController();
+
+    public static ArmSubsystem getInstance() {
+      if (instance == null) {
+        SubsystemChecker.subsystemConstructed(SubsystemType.ArmSubsystem);
+        instance = new ArmSubsystem();
+      }
+      return instance;
+    }
+    public ArmSubsystem() {
+      m_arm = new CANSparkMax(Config.CANID.ARM_SPARK_CAN_ID, motorType); //creates SparkMax motor controller 
+      ErrorCheck.errREV(m_arm.restoreFactoryDefaults());
+      ErrorCheck.errREV(m_arm.setSmartCurrentLimit(Config.ArmConfig.CURRENT_LIMIT));
+      m_arm.setInverted(Config.ArmConfig.SET_INVERTED); //sets movement direction
+      ErrorCheck.errREV(m_arm.setIdleMode(IdleMode.kBrake)); //sets brakes when there is no motion
+      
+      ErrorCheck.errREV(m_arm.setSoftLimit(SoftLimitDirection.kForward, Config.ArmConfig.arm_forward_limit));
+      ErrorCheck.errREV(m_arm.setSoftLimit(SoftLimitDirection.kReverse, Config.ArmConfig.arm_reverse_limit));
+      ErrorCheck.errREV(m_arm.enableSoftLimit(SoftLimitDirection.kForward, Config.ArmConfig.SOFT_LIMIT_ENABLE));
+      ErrorCheck.errREV(m_arm.enableSoftLimit(SoftLimitDirection.kReverse, Config.ArmConfig.SOFT_LIMIT_ENABLE));
+
+      m_arm.setPeriodicFramePeriod(PeriodicFrame.kStatus5, 20);
+      m_arm.setPeriodicFramePeriod(PeriodicFrame.kStatus6, 20);
```

Comment:
Wei was talking about added a comment before those 2 lines explaining that: 
SparkMax periodic status frame 5 is the frequency absolute encoder position data is sent over the can bus.
SparkMax perioidc status frame 6 is the frequency absolute encoder velocity data is sent over the can bus.